projects/rocprofiler-sdk/external/CMakeLists.txt

```
@@ -131,20 +131,24 @@ if(NOT TARGET PTL::ptl-static)
     add_subdirectory(ptl EXCLUDE_FROM_ALL)
 endif()
 
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/yaml-cpp
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/jbeder/yaml-cpp.git
-    REPO_BRANCH "master")
+if(ROCPROFILER_BUILD_YAML_CPP)
+    rocprofiler_checkout_git_submodule(
+        RECURSIVE
+        RELATIVE_PATH external/yaml-cpp
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+        REPO_URL https://github.com/jbeder/yaml-cpp.git
+        REPO_BRANCH "master")
 
-add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
+    add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
 
-target_link_libraries(rocprofiler-sdk-yaml-cpp
-                      INTERFACE $<BUILD_INTERFACE:yaml-cpp::yaml-cpp>)
-target_include_directories(
-    rocprofiler-sdk-yaml-cpp
-    INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/yaml-cpp/include>)
+    target_link_libraries(rocprofiler-sdk-yaml-cpp
+                        INTERFACE $<BUILD_INTERFACE:yaml-cpp::yaml-cpp>)
+    target_include_directories(
+        rocprofiler-sdk-yaml-cpp
+        INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/yaml-cpp/include>)
+else()
+    find_package(yaml-cpp 0.8.0 REQUIRED)
```

```suggestion
    find_package(yaml-cpp 0.8.0 REQUIRED)
    target_link_libraries(rocprofiler-sdk-yaml-cpp
                        INTERFACE $<BUILD_INTERFACE:yaml-cpp::yaml-cpp>)    
```
We "link" to the `rocprofiler-sdk-yaml-cpp` target in rocprofiler-sdk. Even if the find package succeeds when `ROCPROFILER_BUILD_YAML_CPP=OFF`, we will fail to build without it.

---

Good catch.

Even though the build was succeeding for me before with `ROCPROFILER_BUILD_YAML_CPP=OFF`, I have now added the linking command that is done for all other external targets like `GLOG`, `FMT` and `GLOG`:
```
target_link_libraries(rocprofiler-sdk-yaml-cpp INTERFACE yaml-cpp::yaml-cpp)
```